From 7ad1f68dd7631ba309c31fa75ff8c4c82107eb9c Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Thu, 23 May 2013 17:15:21 +0300
Subject: [PATCH 02/34] btwilink: add minimal device tree support

Add minimal device tree support to the btwilink driver that is used
for binding bluetooth with the ti-st shared transport driver.

Change-Id: I301c49d29046f20f8868bebb14347e82c12c8140
Signed-off-by: Eyal Reizer <eyalr@ti.com>
Signed-off-by: bvijay <bvijay@ti.com>
Signed-off-by: varigigi <pierluigi.p@variscite.com>
---
 drivers/bluetooth/btwilink.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/bluetooth/btwilink.c b/drivers/bluetooth/btwilink.c
index f038dba..95adec3 100644
--- a/drivers/bluetooth/btwilink.c
+++ b/drivers/bluetooth/btwilink.c
@@ -30,6 +30,7 @@
 
 #include <linux/ti_wilink_st.h>
 #include <linux/module.h>
+#include <linux/of.h>
 
 /* Bluetooth Driver Version */
 #define VERSION               "1.0"
@@ -286,6 +287,14 @@ static int ti_st_send_frame(struct hci_dev *hdev, struct sk_buff *skb)
 	return 0;
 }
 
+static const struct of_device_id btwilink_of_match[] = {
+{
+	.compatible = "btwilink",
+	},
+	{}
+};
+MODULE_DEVICE_TABLE(of, btwilink_of_match);
+
 static int bt_ti_probe(struct platform_device *pdev)
 {
 	static struct ti_st *hst;
@@ -350,6 +359,7 @@ static struct platform_driver btwilink_driver = {
 	.driver = {
 		.name = "btwilink",
 		.owner = THIS_MODULE,
+		.of_match_table = of_match_ptr(btwilink_of_match),
 	},
 };
 
-- 
2.1.4

